src/module.c

```diff
@@ -10984,6 +10983,14 @@ int RM_IsSubEventSupported(RedisModuleEvent event, int64_t subevent) {
     return 0;
 }
 
+typedef struct {
+    uint64_t version;
+    int32_t dbnum;
+    RedisModuleString *key;
+    robj *value;
+    int mode;
+} KeyInfo;
+
```

Comment:
i understand we need that intermediate struct since we can't open the RedisModuleKey in moduleNotifyKeyUnlink.
i.e. we must do it per module, after we have a context, and that in turn means that we can't create the RedisModuleKeyInfoV1 in moduleNotifyKeyUnlink and just pass it forward in moduleFireServerEvent like we do for REDISMODULE_EVENT_FLUSHDB.

but then, since we fill the actual struct we communicate to the module in moduleFireServerEvent then i think it should be the one to fill the version fields (not moduleNotifyKeyUnlink).